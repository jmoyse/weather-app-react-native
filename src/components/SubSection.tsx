import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';

interface SubSectionProps {
    title?: string
}

interface SubSectionState { }

export default class SubSection extends React.Component<SubSectionProps, SubSectionState> {
    constructor (props: SubSectionProps) {
        super(props);
    }

    render () {
        return (
            <View style={styles.container}>
                {this.props.title && this.props.title.length > 0 ?
                    <Text style={styles.headerText}>
                        {this.props.title}
                    </Text> :
                    <Text></Text>

                }
                {this.props.title && this.props.title.length > 0 ?
                    <View
                        style={styles.line}
                    /> : <View />
                }


                <View style={{ marginBottom: 10 }}>
                </View>
                {this.props.children}
            </View>
        );
    }
}

const low = require('../icons/system/ic_file_download_white_24dp_1x.png');
const high = require('../icons/system/ic_file_upload_white_24dp_1x.png');


const styles = StyleSheet.create({

    container: {
        backgroundColor: '#0000006f',
        alignSelf: 'center',
        padding: 15,
        width: '95%',
        display: 'flex',
        justifyContent: 'space-between',
        flexDirection: 'column',
        marginBottom: 10,
        marginTop: 10,

    },
    headerText: {
        fontSize: 15,
        textAlign: 'left',
        color: 'white',
        marginBottom: 10,

    },
    line: {
        borderBottomColor: '#A8A7A6',
        borderBottomWidth: 1,
    }
});