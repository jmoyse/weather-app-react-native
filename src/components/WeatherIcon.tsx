import * as React from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';

export interface WeatherIconProps {
    weatherID: number;
}

export interface WeatherIconState { }

const blowingsnow = require('../icons/weather/blowing-snow.png');
const cloudy = require('../icons/weather/cloudy.png');
const drizzle = require('../icons/weather/drizzle.png');
const fair = require('../icons/weather/fair.png');
const flurries = require('../icons/weather/flurries.png');
const fog = require('../icons/weather/fog.png');
const freezingrain = require('../icons/weather/freezing-rain.png');
const hazy = require('../icons/weather/hazy.png');
const mcloudynight = require('../icons/weather/m-cloudy-night.png');
const mcloudy = require('../icons/weather/m-cloudy.png');
const pcnight = require('../icons/weather/p-c-night.png');
const pcrain = require('../icons/weather/p-c-rain.png');
const rainysnow = require('../icons/weather/rainy-snow.png');
const rainy = require('../icons/weather/rainy.png');
const showers = require('../icons/weather/showers.png');
const sleet = require('../icons/weather/sleet.png');
const smoke = require('../icons/weather/smoke.png');
const snowshower = require('../icons/weather/snow-shower.png');
const snow = require('../icons/weather/snow.png');
const sunny = require('../icons/weather/sunny.png');
const tstormrain = require('../icons/weather/t-storm-rain.png');
const thunderstorm = require('../icons/weather/thunder-storm.png');
const wind = require('../icons/weather/wind.png');

let json = {
    0: thunderstorm, //'tornado'
    1: tstormrain, //'tropical storm'
    2: thunderstorm, //'hurricane'
    3: thunderstorm, //'severe thunderstorms'
    4: thunderstorm, //'thunderstorms'
    5: rainysnow, //'mixed rain and snow'
    6: sleet, //'mixed rain and sleet'
    7: sleet, //'mixed snow and sleet'
    8: drizzle, //'freezing drizzle'
    9: drizzle, //'drizzle'
    10: freezingrain, //'freezing rain'
    11: showers, //'showers'
    12: showers, //'showers'
    13: flurries, //'snow flurries'
    14: snowshower, //'light snow showers'
    15: blowingsnow, //'blowing snow'
    16: snow, //'snow'
    17: freezingrain, //'hail'
    18: sleet, //'sleet'
    19: hazy, //'dust'
    20: fog, //'foggy'
    21: hazy, //'haze'
    22: smoke, //'smoky'
    23: wind, //'blustery'
    24: wind, //'windy'
    25: fair, //'cold'
    26: cloudy, //'cloudy'
    27: mcloudynight, //'mostly cloudy (night)'
    28: mcloudy, //'mostly cloudy (day)'
    29: pcnight, //'partly cloudy (night)'
    30: pcrain, //'partly cloudy (day)'
    31: sunny, //'clear (night)'
    32: sunny, //'sunny'
    33: fair, //'fair (night)'
    34: fair, //'fair (day)'
    35: rainy, //'mixed rain and hail'
    36: sunny, //'hot'
    37: thunderstorm, //'isolated thunderstorms'
    38: thunderstorm, //'scattered thunderstorms'
    39: thunderstorm, //'scattered thunderstorms'
    40: showers, //'scattered showers'
    41: snow, //'heavy snow'
    42: snowshower, //'scattered snow showers'
    43: snow, //'heavy snow'
    44: pcrain, //'partly cloudy'
    45: thunderstorm, //'thundershowers'
    46: snowshower, //'snow showers'
    47: thunderstorm, //'isolated thundershowers'
};

class WeatherIcon extends React.Component<WeatherIconProps, WeatherIconState> {
    constructor (props: WeatherIconProps) {
        super(props);
    }

    render () {
        return (
            <Image
                style={styles.image}
                source={this.props.weatherID >= 0 ? json[this.props.weatherID] : ''}
            />
        );
    }
}

const styles = StyleSheet.create({
    image: {
        padding: 20,
        height: 30,
        width: 30,
    },
});

export default WeatherIcon;
