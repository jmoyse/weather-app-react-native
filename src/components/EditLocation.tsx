import * as React from 'react';
import { StyleSheet, Text, Image, View } from 'react-native';
import SubSection from './SubSection';
import { store, WeatherAppStore } from '../store/WeatherAppStore';
import { Provider, connect } from 'react-redux';


function mapStateToProps (storeState: WeatherAppStore.Stores): WeatherAppStore.LocationStore {
    return {
        locations: storeState.LocationStore.locations,
    };
}

interface HamburgerMenuProps {
    locations: Array<String>;
}

interface HamburgerMenuState { }

class HamburgerMenuRedux extends React.Component<HamburgerMenuProps, HamburgerMenuState> {
    constructor (props: HamburgerMenuProps) {
        super(props);
    }

    render () {
        return (
            <View style={styles.container} >
                <View style={styles.menuText}>
                    <Text style={styles.headerEntry}>
                        Locations
                    </Text>
                </View>
                <View style={styles.menuEntry}>
                    <Image source={edit_location} style={styles.iconStyle} />
                    <Text style={styles.textStyle} >
                        Edit Locations
                    </Text>
                </View>
                <View style={styles.menuEntry}>
                    <Image source={pin_drop} style={styles.iconStyle} />
                    <Text style={styles.textStyle} >
                        Current Location
                    </Text>
                </View>

                {
                    this.props.locations.map(location =>
                        <View
                            key={Math.random() * 10000}
                            style={styles.menuEntry}>
                            <Image source={pin_drop} style={styles.iconStyle} />
                            <Text style={styles.textStyle} >
                                {location}
                            </Text>
                        </View>
                    )
                }
                <View style={styles.menuText}>
                    <Text style={styles.headerEntry}>
                        Tools
                    </Text>
                </View>
                <View style={styles.menuEntry}>
                    <Image source={pin_drop} style={styles.iconStyle} />
                    <Text style={styles.textStyle} >
                        Settings
                    </Text>
                </View>

                <View style={styles.menuEntry}>
                    <Image source={pin_drop} style={styles.iconStyle} />
                    <Text style={styles.textStyle} >
                        Send Feedback
                    </Text>
                </View>
                <View style={styles.menuEntry}>
                    <Image source={pin_drop} style={styles.iconStyle} />
                    <Text style={styles.textStyle} >
                        Share This App
                    </Text>
                </View>
                <View style={styles.menuEntry}>
                    <Image source={pin_drop} style={styles.iconStyle} />
                    <Text style={styles.textStyle} >
                        Rate this App
                        </Text>
                </View>
            </View>
        );
    }
}

const add_location = require('../icons/system/ic_add_location_white_24dp_1x.png');
const edit_location = require('../icons/system/ic_edit_location_white_24dp_1x.png');
const pin_drop = require('../icons/system/ic_pin_drop_white_24dp_1x.png');

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#272627',
        //padding: 10,
        height: '100%',
        width: '100%'
    },
    menuEntry: {
        width: "100%",
        display: "flex",
        flexDirection: "row",
        paddingLeft: 5,
        alignItems: "center"
    },
    menuText: {
        backgroundColor: "black",
        width: "100%",
        height: 30
    },

    headerEntry: {
        color: "white",
        paddingLeft: 10,
        paddingTop: 5,
    },
    iconStyle: {
        margin: 5
    },
    textStyle: {
        color: "white"
    },
});

const HamburgerMenu = connect(mapStateToProps)(HamburgerMenuRedux);
export default HamburgerMenu;
