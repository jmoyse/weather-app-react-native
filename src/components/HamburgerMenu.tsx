import * as React from 'react';
import { StyleSheet, Text, View, DrawerLayoutAndroid } from 'react-native';
import SubSection from './SubSection';

interface HamburgerMenuProps {

}

interface HamburgerMenuState { }

export default class HamburgerMenu extends React.Component<HamburgerMenuProps, HamburgerMenuState> {
    constructor (props: HamburgerMenuProps) {
        super(props);
    }
    componentWillReceiveProps (newProps: HamburgerMenuProps, newState: HamburgerMenuState) {
        console.log(newProps);
    }

    render () {
        return (
            <SubSection>

                <Text style={{ color: 'white' }}> Work in progress </Text>
            </SubSection>
        );
    }
}

const styles = StyleSheet.create({
    container: {
    },
});
