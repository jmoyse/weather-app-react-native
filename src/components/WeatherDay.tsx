import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import WeatherIcon from './WeatherIcon';

interface WeatherDayProps {
    dayJSON: Object;
}

interface WeatherDayState { }

export default class WeatherDay extends React.Component<WeatherDayProps, WeatherDayState> {
    constructor (props: WeatherDayProps) {
        super(props);

    }

    render () {
        return (
            <View style={styles.container}>
                <Text style={styles.dayText} >
                    {(this.props.dayJSON as any).day}
                </Text>

                <WeatherIcon weatherID={Number.parseInt((this.props.dayJSON as any).code)} />

                <View style={styles.tempsView}>
                    <Text style={styles.highText} >
                        {(this.props.dayJSON as any).high + '°'}
                    </Text>
                    <Text style={styles.lowText} >
                        {(this.props.dayJSON as any).low + '°'}
                    </Text>
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    lowText: {
        fontFamily: 'HelveticaNeueLTStd_Th',
        fontSize: 15,
        color: '#8FBFE8'
    },
    highText: {
        fontFamily: 'HelveticaNeueLTStd_Th',
        fontSize: 15,
        color: 'white',
        paddingRight: 15
    },
    dayText: {
        fontFamily: 'HelveticaNeueLTStd_Th',
        fontSize: 15,
        textAlign: 'center',
        color: 'white',
        width: 100
    },
    tempsView: {
        display: 'flex',
        flexDirection: 'row',
    },
    container: {
        //paddingTop: 5,
        //paddingLeft: 10,
        //paddingRight: 10,

        //height: 150,

        display: 'flex',
        flexDirection: 'row',
        //width: '100%',
        alignContent: 'center',
        alignItems: 'center',
        //alignItems: 'center',
        justifyContent: 'space-between',
    },
});
