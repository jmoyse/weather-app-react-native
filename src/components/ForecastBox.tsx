import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import WeatherIcon from './WeatherIcon';
import WeatherDay from './WeatherDay';
import SubSection from './SubSection';

interface ForecastBoxProps {
    forecastJson: Object;
}

interface ForecastBoxState { }

export default class ForecastBox extends React.Component<ForecastBoxProps, ForecastBoxState> {
    constructor (props: ForecastBoxProps) {
        super(props);
    }
    componentWillReceiveProps (newProps: ForecastBoxProps, newState: ForecastBoxState) {
        console.log(newProps);
    }

    render () {
        return (

            <View style={styles.container}>
                {
                    (this.props.forecastJson as any) ?
                        (
                            (this.props.forecastJson as any).map(dayNode =>
                                <WeatherDay key={((dayNode as any).date as string)} dayJSON={dayNode} />
                            )
                        ) : <Text> N/A </Text>

                }
            </View>

        );
    }
}

const styles = StyleSheet.create({
    container: {
        //backgroundColor: '#0000007f',
        //margin: 10,
        //padding: 10,
    },
});
