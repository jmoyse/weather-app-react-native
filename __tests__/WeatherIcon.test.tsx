import React from 'react';
import 'react-native';
import { WeatherIconProps, WeatherIcon, icons, BLANK_ID } from '../src/components/WeatherIcon';
import { shallow, configure, mount, ShallowWrapper, render, } from 'enzyme';

describe("WeatherIcon", () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = { weatherID: 0 };
    wrapper = shallow(<WeatherIcon {...props} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders expected image for weatherID 0', () => {
    wrapper.setProps({ weatherID: 0 });
    wrapper.render();
    let iconSource = wrapper.find("Image").prop("source");
    expect(iconSource).toEqual(icons[0]);
  });

  it('renders expected update image for weatherID 25', () => {
    wrapper.setProps({ weatherID: 25 });
    let iconSource = wrapper.find("Image").prop("source");
    expect(iconSource).toEqual(icons[25]);
  });

  it('renders expected invalid image', () => {
    wrapper.setProps({ weatherID: -1 });
    let iconSource = wrapper.find("Image").prop("source");
    expect(iconSource).toEqual(icons[BLANK_ID]);
  });
});
