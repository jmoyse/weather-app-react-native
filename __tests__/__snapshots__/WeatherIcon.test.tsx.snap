// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherIcon matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherIcon
    weatherID={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "source": "13.png",
      "style": Object {
        "flex": 1,
        "height": undefined,
        "padding": 15,
        "width": undefined,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "source": "13.png",
        "style": Object {
          "flex": 1,
          "height": undefined,
          "padding": 15,
          "width": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
